### PR TITLE
Resolved 'Not' Operator Issue in Binary Node Expression

### PR DESCRIPTION
**Pull request by ckganesan**
This pull request addresses issue #589.

### CODE REVIEW

File: parser/parser.go [hunk 1]
```diff
@@ -126,48 +126,48 @@ func (p *parser) expect(kind Kind, values ...string) {
...
+		return p.parseVariableDeclaration()
 	}
 
 	nodeLeft := p.parsePrimary()
 
 	prevOperator := ""
-	opToken := p.current
-	for opToken.Is(Operator) && p.err == nil {
-		negate := false
+	for opToken := p.current; opToken.Is(Operator) && p.err == nil; opToken = p.current {
```
Comment (antonmedv) [inline parser/parser.go:136]: I think it's better to keep the old formatting, as we iterating not only on `opToken`, but also updating `prevOperator` on each cycle.

File: parser/operator/operator.go [hunk 2]
```diff
@@ -20,6 +20,15 @@ func IsBoolean(op string) bool {
 	return op == "and" || op == "or" || op == "&&" || op == "||"
 }
 
+func IsFunc(op string) bool {
```
Comment (antonmedv) [inline parser/operator/operator.go:23]: I know we have a PR with this func in separate PR. But for this PR lets rename the func to something like:  `AllowNegatePrefix`

File: parser/parser_test.go [hunk 3]
```diff
@@ -365,6 +365,62 @@ world`},
...
+					},
+					Right: &ArrayNode{Nodes: []Node{
+						&IntegerNode{Value: 1},
+						&IntegerNode{Value: 2},
+						&IntegerNode{Value: 3},
+						&IntegerNode{Value: 4},
+					}}}},
+		},
+		{
+			"2==2 ? false : 3 not in [1, 2, 5]",
```
Comment (antonmedv) [inline parser/parser_test.go:396]: Will be cool to require a brackets `(` `)` in this case too. As even for me, it's unclear what operator will be applied first.

Comment (antonmedv) [reply parser/parser_test.go:396]:
```
2 == 2 ? false : (3 not in [1, 2, 5])
```

But we can implement it in a new PR.

File: parser/parser_test.go [hunk 4]
```diff
@@ -706,6 +762,11 @@ invalid float literal: strconv.ParseFloat: parsing "0o1E+1": invalid syntax (1:6
 invalid float literal: strconv.ParseFloat: parsing "1E": invalid syntax (1:2)
  | 1E
  | .^
+
+1 not == [1, 2, 5]
+unexpected token Operator("==") (1:7)
```
Comment (antonmedv) [inline parser/parser_test.go:767]: Nice! I like this test.

File: parser/parser.go [hunk 5]
```diff
@@ -126,48 +126,48 @@ func (p *parser) expect(kind Kind, values ...string) {
 // parse functions
 
 func (p *parser) parseExpression(precedence int) Node {
-	if precedence == 0 {
-		if p.current.Is(Operator, "let") {
-			return p.parseVariableDeclaration()
-		}
+	if precedence == 0 && p.current.Is(Operator, "let") {
```
Comment (antonmedv) [inline parser/parser.go:129]: Nice catch 👍🏻

**Review by antonmedv — CHANGES_REQUESTED** [review]

File: compiler/compiler_test.go [hunk 6]
```diff
@@ -541,6 +541,48 @@ func TestCompile_optimizes_jumps(t *testing.T) {
...
+			`-1 not in [1, 2, 5]`,
+			[]op{
+				{vm.OpPush, 0},
+				{vm.OpPush, 1},
+				{vm.OpIn, 0},
+				{vm.OpNot, 0},
+			},
+		},
+		{
+			`-1 not in [1, 2, 5]`,
```
[the remark this answers is not in the document]
Comment (ckganesan) [reply compiler/compiler_test.go]: Duplicate removed based on your feedback.

File: parser/parser.go (hunk 1 above)
Comment (ckganesan) [reply parser/parser.go]: Reverted to old format as per your feedback.

File: parser/operator/operator.go (hunk 2 above)
Comment (ckganesan) [reply parser/operator/operator.go]: Renamed function name as "AllowedNegateSuffix"

File: parser/parser_test.go (hunk 3 above)
Comment (ckganesan) [reply parser/parser_test.go:396]: This behavior aligns with expectations since the ':' operator is not considered a comparison operator.

**Pull request merged**